File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math
from collections import deque

def solution(progresses, speeds):
    result = []

    progresses = deque(progresses)

    for i in range(len(progresses)):
        progresses[i] = (100 - progresses[i]) / speeds[i]
        progresses[i] = math.ceil(progresses[i])

    while progresses:

        if progresses:
            compare = progresses.popleft()
        cnt = 1
        while progresses:
            if progresses[0] <= compare:
                cnt += 1
                progresses.popleft()
            else:
                break
        result.append(cnt)

    return result

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.LinkedList;
import java.util.Queue;

public class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        LinkedList<Integer> result = new LinkedList<>();
        Queue<Integer> queue = new LinkedList<>();

        // Calculate days needed to finish each progress and push to queue
        for (int i = 0; i < progresses.length; i++) {
            int days = (100 - progresses[i] + speeds[i] - 1) / speeds[i];
            queue.offer(days);
        }

        while (!queue.isEmpty()) {
            int current = queue.poll();
            int count = 1;

            while (!queue.isEmpty() && queue.peek() <= current) {
                count++;
                queue.poll();
            }

            result.add(count);
        }

        // Convert LinkedList to array
        return result.stream().mapToInt(i -> i).toArray();
    }
}